### ISSUE TITLE
Popover arrow is not positioned correctly when given 2 positionals

### ISSUE THREAD

**Issue by zeroasterisk** [reactstrap/reactstrap#114]
```
Given a standard popover `Popover` 
And it is positioned with 2 positions (eg: `bottom right`)
When it is opened
Then it is positions correctly
And it's arrow is not positioned correctly
```
- version `reactstrap@1.6.1` & `bootstrap@4.0.0-alpha.2` & `tether@1.3.4`
- components: `Popover`

![ss](https://puu.sh/qXC6t/0dd6291320.png)

NOTE: I am not using `bootstrap@4.0.0-alpha.3` because their npm release is broken and I did not choose to manually install (yet).

(this might be a bootstrap thing)

**Comment by eddywashere** [comment]
A couple of comments, but most importantly, I’m excited you are also interested in advanced positioning of popovers! Bootstrap css currently [supports 4 directions](http://v4-alpha.getbootstrap.com/components/popovers/#four-directions): top, left, right, bottom. However, this is not enforced by reactstrap [:\]](https://github.com/react-bootstrap/react-bootstrap/blob/f30cc470463af98eedd37b945deb46a784ded738/src/Popover.js#L21)

I originally set out to add the css for stuff like this to a separate project, similar to [mgcrea/bootstrap-additions](https://github.com/mgcrea/bootstrap-additions) but I have not had time. It would be great if bootstrap added it. Their js seems to also support it.

I first came across this in [ngstrap](http://mgcrea.github.io/angular-strap/#/popovers). Definitely part of the inspiration for reactstrap and some of it's opinions.

![image](https://d3g7hm1jmpcdis.cloudfront.net/v2zcs0yv3b.png)

I’ll keep this issue open until there is a decent solution we can recommend. Thanks!

Also, for codepen demo purposes, I’ve been using unpkg.com (previously npmcdn.com). You could use this - https://unpkg.com/bootstrap@4.0.0-alpha.3/dist/css/bootstrap.min.css

**Comment by TheSharpieOne** [comment]
Just adding this link with a demo of all of the different placements mentioned in the docs: http://codepen.io/TheSharpieOne/full/KgKxXy/

Note: the docs show the PropTypes for Popover and the placement proptype specifically allows for all of the combinations.

**Comment by zeroasterisk** [comment]
Just checking back in on this one --- any ideas?  

I can "fix" this in the inspector by:

1. remove the class name `bs-tether-element-attached-left` from the parent
2. put `left: 10%' into the arrow element styles

![ss](https://puu.sh/rGuhP/7a89a45424.png)

The arrow looks funky because both the `-left` and the `-bottom` are affecting `:after` to make the arrow.

The positioning is funky for approx the same reason.

I think this is a CSS fix for the combos of side+top (corner) positions.

**Comment by TheSharpieOne** [comment]
Yes, I think CSS is needed to fix this. The problem/issue is: Should this project introduce CSS?
Currently, it's 100% components, 0% styles. All of the styles come from bootstrap (and potentially, the user of the project's own/custom styles).
Bootstrap doesn't support the additional combinations and fixing it would be pretty easy, but is it outside of the scope of this project?

**Comment by LiJinyao** [comment]
Maybe we should remove the extra position?

**Comment by eddywashere** [comment]
This project itself should not introduce the css. The extra positions that are included, which are not supported in bootstrap css, are there as a convenience. I'm going to close this as it's not really actionable for this project. I still personally plan to tackle this in my free time. I'm not a fan of the default positions : ] and haven't had to introduce it just yet. When I do, @zeroasterisk will be the first person I ping. Last time I looked, I think tether has different classes, and one would need to port [these](https://github.com/mgcrea/bootstrap-additions/blob/master/dist/bootstrap-additions.css#L336-L354) over to the tether classes attached to popover/tooltip.

**Comment by Geczy** [comment]
Have this issue as well, shoudn't this be open? 

I came from #258

**Comment by TheSharpieOne** [comment]
It was closed because the extra positions are not supported by bootstrap. You can add custom CSS to handle the extra positions.

**Comment by gthomas-appfolio** [comment]
I think this is confusing because the code and documentation lists the unsupported 'extra' positions as options: https://reactstrap.github.io/components/popovers/
and this has tripped up some people who think this then should 'just work'.

If we want to leave this flexible for addons, etc.  can we just list the `placement: PropTypes.string` ?

**Comment by eddywashere** [comment]
That's a bummer. Don't feel too strict on this, and am definitely for improving ux. I'd +1 a change to a generic string and updating docs to show `// ex: 'top', 'right', 'bottom', 'left'`.

**Comment by gthomas-appfolio** [comment]
Thanks @eddywashere , I can make a PR

**Comment by turnerhayes** [comment]
So to make sure I understand; the PR for this won't be supporting the multiply positions, just clarifying that they're not supported in the docs, right?

**Comment by gthomas-appfolio** [comment]
@turnerhayes Yes, that was my intent
